Add option to control the order of the days in the `log` output

joelostblom opened the pull request:
I think it is quite common to sort with the newest entry at the bottom (e.g with `ls` and this is also how the times are ordering within each day in the `log` output), so I suggest adding an option to enable this behavior.

joelostblom commented:
Will add a flag with the same name `--log_latest_day_on_top`!

joelostblom commented:
That's what I had initially, but then I wanted to be more specific to indicate that it is only reversing the order of the days, not the time entries within the days, so that I would leave it open for someone to add an option to reverse times if desired. This might never be the case though, and I am happy to rename both the option and the flag `reverse-log` if you prefer that.

jmaupetit commented:
`--reverse` should be fine, thanks!

joelostblom commented:
I can try, but might need some assistance to make it satisfactory. Before giving it a go, I just want to clarify what what you would like this test to contain. Do you want it to be something simple like the [current CLI log tests](https://github.com/TailorDev/Watson/blob/af8911715ce131218c413751197b6f0ce1882b5b/tests/test_cli.py#L131-L150), just adding the `--reverse` flag and check that no errors are raised? Or do you want there to be a new `log` section in `test_watson.py` which checks the actual ordering of the days when using `--reverse/--no-reverse`?

jmaupetit commented:
Testing Watson's API is easier to implement, so adding a new log section in `test_watson.py` would be perfect.

joelostblom commented:
@jmaupetit  Let me know what you think of the tests I just added. Since `log` is defined entirely in `cli.py`, I could not figure out a way to test it in `watson.py`. I tried adding it in with the other cli-tests but could not figure out a good way to add data and test it, so I added the tests to `test_utils.py`. The disadvantage is that we're really only testing the inner most function and things could of course go wrong outside, but I wasn't able to implement anything better. Let me know if you have suggestions. The failing flake8test is unrelated and fails on master as well.

joelostblom commented:
@jmaupetit Just a reminder to have a look at this when you get a chance.

jmaupetit commented:
Thanks @joelostblom ! If you handle the rebase, I'll merge this asap :pray:

joelostblom commented:
Haha thanks! =) I just rebased and squashed a few of the commits.

joelostblom commented:
Realized that the changelog entry was in the wrong place since the 1.9 tag was added recently. Ready for merge now :+1:
